sr_piezo_feedback/src/sr_piezo_feedback_node.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import
+from __future__ import division
+import argparse
+import sys
+import math
+import rospy
+import numpy as np
+import sounddevice as sd
+import matplotlib.pyplot as plt
+from sr_robot_msgs.msg import ShadowPST, BiotacAll
+from std_msgs.msg import Float64, Float64MultiArray, Header
+import threading
+import time
+from sr_hand.tactile_receiver import TactileReceiver
+import matplotlib.pyplot as plt
+
+from dynamic_reconfigure.server import Server
+from sr_piezo_feedback.cfg import SrPiezoFeedbackConfig
+from sr_piezo_feedback.msg import PiezoFeedback
+
+
+class DeviceHandler(threading.Thread):
+    def __init__(self, device, fingers, mount):
+        super(DeviceHandler, self).__init__()
+        self._device_name = device
+        self._finger_per_devices = fingers
+        self._mount = mount
+        self._samplerate = sd.query_devices(self._device_name, 'output')['default_samplerate']
+
+        self._ts = [0] * len(self._finger_per_devices)
+        self._oldsignal = [0] * len(self._finger_per_devices)
+        self._freq = [1] * len(self._finger_per_devices)
+        self._amp = [1] * len(self._finger_per_devices)
+        self._signal_range = [-0.1, 0.6]
+        self._offset = np.mean(self._signal_range)
+        self._amp_factor = np.mean(np.abs(self._signal_range))
+
+        self._publisher = dict(zip(self._finger_per_devices, len(self._finger_per_devices) * [0]))
+        self._finger_msg = dict(zip(self._finger_per_devices, len(self._finger_per_devices) * [0]))
+
+        for finger in self._finger_per_devices:
+            self._publisher[finger] = rospy.Publisher('sr_piezo_feedback/'+finger, PiezoFeedback, queue_size=1)
+            self._finger_msg[finger] = PiezoFeedback()
```

Comment:
I think here we should have the: self._used_tactiles = TactileReceiver(self._hand_id).get_tactile_type()
Then based on this we should init the proper class. PiezoFeedback should be parent class of PiezoFeedbackPST() and PiezoFeedbackBiotac().
Otherwise you end-up with a class with lots fo things happening and with low reusability. For instance, if, in the future, you want to add PiezoFeedbackMST(), you will be able to just create a new class with 2 simple functions.